pkg/code-generator/schemagen/README.md

```diff
@@ -0,0 +1,49 @@
+# Schemagen
+A tool used to generate validation schemas for Kubernetes CRDs
+
+## Validation Schemas
+Custom Resources in Kubernetes support [defining a structural schema](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema), which is validated during creation and updates. V1 CRDs (required in Kube 1.22 and beyond) require validation schemas. These schemas provide syntax validation (ie does the configuration respect the API structure).
+
+## Options
```

Comment:
Could use more explanation for why/when these options might be used (apart from `MessagesWithEmptySchema`

Reply:
Which options would you like more clarity on? I'm happy to make them clearer, and perhaps I've been using them too much, but I can't think of ways to clarify this further

Reply:
Why might we want to use a `jsonSchemaTool` other than `protoc`? or remove descriptions from validation schemas? or allow enums to be ints or strings?

Reply:
How is https://github.com/solo-io/solo-kit/pull/459/commits/097c7b6b9756fce884a83034af880eccb372fd1a?

Reply:
👍